### PR TITLE
Set default "n_threads" value to half of the available threads

### PR DESCRIPTION
**Pull request by ryan-mca**
Just a QoL type thing.

Using psutil.cpu_count() and then dividing it by 2

Tested this on my own machine and it seemed to work fine

From seperate testing if the thread count is 1 it should just set the default value to 0

### CODE REVIEW

File: api/main.py [hunk 1]
```diff
@@ -105,6 +106,7 @@ def list_of_installed_models(
 ):
     return models
 
+THREADS = cpu_count()
```
Comment (gaby) [inline api/main.py:109]: We should only use the number of assigned cores, not the physical ones. 

This can be done with:

```
THREADS = len(psutil.Process().cpu_affinity())
```

File: api/main.py [hunk 2]
```diff
@@ -117,7 +119,7 @@ async def create_new_chat(
     repeat_last_n: int = 64,
     repeat_penalty: float = 1.3,
     init_prompt: str = "Below is an instruction that describes a task. Write a response that appropriately completes the request. The response must be accurate, concise and evidence-based whenever possible. A complete answer is always ended by [end of text].",
-    n_threads: int = 4,
+    n_threads: int = THREADS / 2,
```
Comment (gaby) [inline api/main.py:122]: This should be `n_threads = THREADS`, in case it's 1/2

**Review by gaby — CHANGES_REQUESTED** [review]

File: api/main.py (hunk 1 above)
Comment (ryan-mca) [reply api/main.py]: Just fixed that, thanks for the suggestion

File: test.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+num = int(1 / 2)
```
Comment (gaby) [inline test.py:1]: Is this file needed?

File: api/main.py (hunk 2 above)
Comment (gaby) [reply api/main.py:122]: There's a chance the user wants to use all threads available, which this would limit.

Comment (gaby) [reply api/main.py:122]: @nsarrazin thoughts on this?

Comment (nsarrazin) [reply api/main.py:152]: I think it's fine, it's only shown in the API anyway. This doesn't update the front-end  value which is still 4... But still a good idea. 

I'm not sure why it should be threads/2 though. `llama.cpp` by default uses the total number of threads available so I think we might as well just copy what they do.

File: api/requirements.txt [hunk 4]
```diff
@@ -40,4 +41,4 @@ urllib3==1.26.15
 uvicorn==0.21.1
 uvloop==0.17.0
 watchfiles==0.18.1
-websockets==10.4
+websockets==10.4
```
Comment (nsarrazin) [inline api/requirements.txt:44]: Why do we need websockets ?

Comment (ryan-mca) [reply api/requirements.txt:44]: That was already there, I put the psutil import at the bottom originally and fixing that made it look like I added it

File: api/main.py (hunk 2 above)
Comment (ryan-mca) [reply api/main.py:152]: Could there be anywhere else that could change it at the front-end? I saw that there was an n_threads that sets the default value in models.py, but that should get overwritten in main.py?

**Comment by gaby** [comment]
@nsarrazin I think this can be merged.
